### ISSUE TITLE
[BUG] Errors & unexpected behaviour for fields names which are reserved keywords.

### ISSUE THREAD

**Issue by dnx-aiven** [opensearch-project/sql#1589]
**What is the bug?**

tl;dr, reserved keywords cause PPL search issues & UI bugs when a field is of these names.

I have various fields, which I can utilise in DQL searches. However, in Observability (I'm working in Event Analytics section), the fields display in UI, but the values for these are not populated, and when querying with PPL there are errors.

I have tested this on OS 2.4.1, which is my install. I tried the latest in the playground, but that wasn't working well for me at all.
The sample commerce data has a field named `type` so this is a good one to use for testing.

The bugs are not present in my 1.3.8 install.

There is a workaround for PPL, which is that field names which are reserved keywords need to be backticked. If I use quotes there isn't an error, but there aren't any results either.

For the UI, there's some bug which needs addressing.
 
Reserved keywords are not documented anywhere that I can see. I know `type` and `ip`. What are others?

Needing to use backticks for some field names is not documented. I feel like this is a workaround rather than a fix.

Error messages are not very helpful. Could be more accurate as to the specific issue.

**How can one reproduce the bug?**
1. Go to Event Analytics
2. Search a source where the index has a field of the same name as a restricted keyword.
3 Observe no values populated for the field
4 Add a search such as `| where type = "value"`
5 Observe error.

**What is the expected behavior?**
UI to populate field values.
PPL to not need backticks, or for this to be documented.
Documentation coverage.

**What is your host/environment?**
 - OS: Fedora. Also tried via docker-compose.
 - Version 2.4.1
 - Plugins: Just stock plugins. 

**Do you have any screenshots?**
![image](https://user-images.githubusercontent.com/103011565/234778225-e57b8449-08eb-4142-945a-5f36218ca8a0.png)

![image](https://user-images.githubusercontent.com/103011565/234778261-e36e6043-cdc4-4183-8c94-809f3b50e170.png)

![image](https://user-images.githubusercontent.com/103011565/234778273-9dd60f7d-a260-4a8a-98a4-f92a99388259.png)

**Do you have any additional context?**

As discussed with @anirudha & others from the team.

There is a previous Issue raised for this with a field named `ip` https://github.com/opensearch-project/observability/issues/1148
IMHO it was closed with too little information, due to restricted fields not being documented and the unhelpful error messages.

**Comment by Yury-Fridlyand** [comment]
We should extend `keywordsCanBeId` list in grammar file (it was partially done in #1319).
As a workaround you can escape identified with backticks `` ` ``, e.g. ``... | where `type` = "value"``

**Comment by dnx-aiven** [comment]
I've tried again today to use Observability and getting errors such as this. I presume it's from the same bug, as `type` field exists in this index. I'd like to update all our index fields. Is a list of all reserved keywords available anywhere?

<img width="833" alt="image" src="https://github.com/opensearch-project/sql/assets/103011565/b3bfd1dd-b970-4624-986c-737ea86f43c6">

**Comment by Yury-Fridlyand** [comment]
All PPL keywords are listed here: https://github.com/opensearch-project/sql/blob/main/ppl/src/main/antlr/OpenSearchPPLLexer.g4, but ones are not in `keywordsCanBeId` list should be wrapped by backticks to avoid issues.
https://github.com/opensearch-project/sql/blob/0dfbc892c3f6f34914661122dcaf04a6a703949f/ppl/src/main/antlr/OpenSearchPPLParser.g4#L829C1-L857

**Comment by dnx-aiven** [comment]
Thank you @Yury-Fridlyand . backticks help with search, but they do not help with the UI issues encountered with those fields.
